src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/HKDF.cs

```diff
@@ -108,12 +113,17 @@ public static byte[] Expand(HashAlgorithmName hashAlgorithmName, byte[] prk, int
         /// </summary>
         /// <param name="hashAlgorithmName">The hash algorithm used for HMAC operations.</param>
         /// <param name="prk">The pseudorandom key of at least <see cref="HashLength"/> bytes (usually the output from Expand step).</param>
-        /// <param name="output">The destination buffer to receive the output keying material.</param>
+        /// <param name="output">The destination buffer to receive the output keying material with a size of as least 1.</param>
         /// <param name="info">The context and application specific information (can be an empty span).</param>
+        /// <exception cref="ArgumentException"><paramref name="output"/> length is larger than the allowed length.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="output"/> legnth is less than 1.</exception>
         public static void Expand(HashAlgorithmName hashAlgorithmName, ReadOnlySpan<byte> prk, Span<byte> output, ReadOnlySpan<byte> info)
         {
             int hashLength = HashLength(hashAlgorithmName);
 
+            if (output.Length == 0)
+                throw new ArgumentOutOfRangeException(nameof(output), SR.ArgumentOutOfRange_NeedPosNum);
```

Comment:
The output is too short, but we can't really say it needs to be a positive number.  There's a good enough message available, though:

```suggestion
                throw new ArgumentOutOfRangeException(nameof(output), SR.Argument_DestinationTooShort);
```

Reply:
Should probably change the exception type to `ArgumentException` for the "filling existing buffers" exceptions and keep `ArgumentOutOfRangeException` for the "takes a length" exceptions?

Reply:
Whoops, nice catch.  Updated my code suggestion.

Reply:
One other thing: the order of the parameters between `ArgumentOutOfRangeException` and `ArgumentException`. 

One of my "favorites".

Reply:
I swapped the exceptions produced for `Span<byte>` to be `ArgumentException` with the correct error message.